### DISCUSSION TITLE
Introducing 'expireTime' on `MsgCreateAllianceProposal`

### DISCUSSION THREAD

**Discussion by emidev98** [terra-money/alliance#268] [Ideas]
### Feature

To enhance the utility of the Alliance module, I propose to introduce an optional field called `expireTime` for the `MsgCreateAllianceProposal`. This field indicates the specific block time at which the alliance will be deactivated, stakers will stop receiving rewards and staked tokens will start their unbonding period. This feature encourages teams to create alliances with their tokens for a specific duration beforehand. 

### Example of use cases

- Create an alliance with an [LP](https://www.coindesk.com/learn/what-are-liquidity-pools/) token pair to boost LP liquidity. Liquidity providers will have an incentive to stake their LP token in the alliance module to receive staking rewards from the blockchain native token for a specific time that is voted in by the chain governance. When liquidity providers stake their LP token they are subjected to the unbonding period of the chain which means that they trade that LP token for yield during the stake of the token.

```mermaid
stateDiagram-v2
    LUNA --> LiquidityPool: Deposit in DEX
    USDC --> LiquidityPool: Deposit in DEX
    LiquidityPool --> LiquidityPoolToken: receive
    LiquidityPoolToken --> MsgStake: stake token in alliance
```

- Use [Alliance Protocol](https://github.com/terra-money/alliance-protocol) or similar smart contract to incentive to low market cap tokens to attract holders because Alliance Protocol allows unstaking without unbonding period.

```mermaid
stateDiagram-v2
    LUNA --> LiquidityPool: Deposit in DEX
    USDC --> LiquidityPool: Deposit in DEX
    LiquidityPool --> LiquidityPoolToken: receive
    LiquidityPoolToken --> AllianceProtocolContract: execute msg alliance delegate
    AllianceProtocolContract --> MsgStake: staked virtual tokens in alliance module
```

For more details on the use cases of the Alliance module, you can refer to [this link](https://docs.alliance.terra.money/overview/#use-cases).

> [Look into how alliance assets can be removed and delegations returned](https://github.com/terra-money/alliance/issues/94)

**Comment by Mrt3383x38** [comment]
Binance giriş eror

**Reply by Mrt3383x38** [reply]
![Screenshot_20231108_223029_Binance.jpg](https://github.com/terra-money/alliance/assets/117018255/032a3cbe-0ba7-40d0-9cc3-3dc109e16871)

Giriş eror

**Comment by Mrt3383x38** [comment]
![Screenshot_20231114_014741_Chrome.jpg](https://github.com/terra-money/alliance/assets/117018255/e32671ba-de42-4611-b5fd-a57d136b8373)
